0026-an-extensible-db-design.adoc

```diff
@@ -0,0 +1,283 @@
+
+* `han_bitcoin/ethereum_swap_detail` contains data needed to create the HTLCs. We decided to give this table a generated primary key, rather than using `swap_local_id` as the primary key to allow same chain swaps.
+* `han_bitcoin_ethereum_redeem/refund_identity`: Since some identities are exchanged during announcement, the identities are kept in extension tables.
+
+HErc20 and HaLight follow the same principles as described above.
+For completeness we depict `herc20` and `halight` diagrams below.
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/han_ethereum_swap_protocols_without_state.puml&fmt=svg[HanEthereumProtocols]
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/herc20_swap_protocols_without_state.puml&fmt=svg[Herc20Protocols]
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/halight_swap_protocols_without_state.puml&fmt=svg[HalightProtocols]
+
+
+==== Protocols with states:
+
+* `han_bitcoin/ethereum_ledger_state`: is a new table which we introduced in this database schema.
+It records observed ledger events such as `fund`/`redeem`/`refund`-transaction.
+We extracted the states into `han_bitcoin_state` in order to reduce duplication in each table and not have an extra column for the event-type (`fund`/`redeem`/`refund`).
+We could reuse this table across all `han_*` protocols but decided for separate _state_ tables per protocol.
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/han_bitcoin_swap_protocols_with_state.puml&fmt=svg[HanBitcoinProtocols]
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/han_ethereum_swap_protocols_with_state.puml&fmt=svg[HanEthereumProtocols]
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/herc20_swap_protocols_with_state.puml&fmt=svg[Herc20Protocols]
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/halight_swap_protocols_with_state.puml&fmt=svg[HalightProtocols]
+
+=== Swap State
+
+image::http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/comit-network/spikes/86bfc97bcea91602eaf3e40e6da7d71beaff9911/assets/db_schema/swap_states.puml&fmt=svg[SwapStates]
+
+Additional to the specific `alpha` and `beta` ledger state (depicted in <<Protocols>>) a `swap` has two additional states:
```

Comment:
Same for these states. If you want to include them, I would have some more questions but I don't want to start the discussion if we agree on omitting them in the implementation for now.